tests/Doctrine/Tests/DBAL/Functional/ConnectionTest.php

```diff
@@ -196,11 +196,30 @@ public function testTransactionalWithException()
                 $conn->executeQuery($conn->getDatabasePlatform()->getDummySelectSQL());
                 throw new \RuntimeException("Ooops!");
             });
+            $this->fail('Expected exception');
         } catch (\RuntimeException $expected) {
             $this->assertEquals(0, $this->_conn->getTransactionNestingLevel());
         }
     }
 
+    public function testTransactionalWithThrowable()
+    {
+        if (version_compare(PHP_VERSION, '7.0', '<')) {
```

Comment:
Yeah I did, and it failed hard https://github.com/doctrine/dbal/pull/2390#discussion-diff-63157912